### ISSUE TITLE
Creating multiple search index records per model?

### ISSUE THREAD

**Issue by abm-devmonk-purple** [algolia/scout-extended#212]
Sometimes there are cases where multiple, customized search index records need to be created per model record. As far as I know, scout-extended only supports customizing attributes for a single search index record via the `toSearchableArray()` method, or morphing one or more of these attributes via the `split*()` methods.

I added https://github.com/algolia/scout-extended/pull/211 as a possible solution, but I'm not sure if that's the best way to handle this situation.

I found a related issue here https://github.com/laravel/scout/issues/275.

**Comment by nunomaduro** [comment]
Can you give me your use case for this feature? Let's first try to see if can be done with the `split*`.

**Comment by abm-devmonk-purple** [comment]
To simplify our situation, we have models that contain attributes over two languages. So it looks like this:

```php
Foo::create(['title_en' => 'one', 'title_th' => 'หนึ่ง' ]);
Foo::create(['title_en' => 'two', 'title_th' => 'สอง' ]);
```

We would like to be able to output the following search index records:

```php
$records = [
    [ 'lng' => 'en', 'title' => 'one', ... ],
    [ 'lng' => 'th', 'title' => 'หนึ่ง', ... ],
    [ 'lng' => 'en', 'title' => 'two', ... ],
    [ 'lng' => 'th', 'title' => 'สอง', ... ],
];

We weren't sure if there was a way forward with the `split*` methods.

**Comment by nunomaduro** [comment]
Thanks,

Let's try something:

1. Using `toSearchableArray`, create a **new** `title` field that contains an array with the following content:
```
[
    'fr'   => $model->title_th,
    'en' => $model->title_en
]
```

2. Then implement a method `splitTitle` that will split the given record based on that the title.

Let me know if this works for you.

**Comment by abm-devmonk-purple** [comment]
Hi there,

Thanks for writing back. We just tried the nested approach you described and it worked! The code change we used is here: https://github.com/abhayagiri/abhayagiri-website/commit/90f8812299c08b74c022764b00fcde4398a1ce70

From the initial testing just a week ago, I recall not being able do certain queries or de-duplication using nested attributes. Did something change recently?

**Comment by nunomaduro** [comment]
@abm-devmonk-purple Can you rephrase your question? I didn't understood.

**Comment by abm-devmonk-purple** [comment]
@nunomaduro I recall a couple weeks ago not being to use `'attributeForDistinct' => 'text.path'`, only something like `'attributeForDistinct' => 'path'`. Was I imagining this?

**Comment by nunomaduro** [comment]
Ha yes! You may need to use `distinct` and `attributeForDistinct` for your search do not return duplicated results, here is the doc about that: https://www.algolia.com/doc/framework-integration/laravel/advanced-use-cases/split-large-records/?language=php#distinct.
